calyx-py/calyx/builder.py

```diff
@@ -271,26 +305,35 @@ def asgn(self, lhs: ExprBuilder,
 
         if isinstance(rhs, int):
             width = infer_width(lhs)
-            assert width, f'could not infer width for literal {rhs}'
+            assert width, f"could not infer width for literal {rhs}"
             rhs = const(width, rhs)
 
         wire = ast.Connect(
             ExprBuilder.unwrap(lhs),
             ExprBuilder.unwrap(rhs),
             ExprBuilder.unwrap(cond),
         )
-        self.group.connections.append(wire)
+        if self.group is not None:
+            self.group.connections.append(wire)
+        else:
+            self.comp.component.wires.append(wire)
 
     @property
     def done(self):
         """The `done` hole for the group."""
-        return ExprBuilder(
-            ast.HolePort(ast.CompVar(self.group.id.name), "done")
-        )
+        if not self.group:
+            raise Exception(
+                "GroupBuilder represents continuous assignments and does not have a done hole"
+            )
+        return ExprBuilder(ast.HolePort(ast.CompVar(self.group.id.name), "done"))
 
     @done.setter
     def done(self, expr):
         """Build an assignment to `done` in the group."""
+        if not self.group:
+            raise Exception(
+                "GroupLikeBuilder represents continuous assignments and does not have a done hole"
+            )
```

Comment:
I'd probably just make this:

```py
assert self.group, "GroupLikeBuilder represents continuous assignments and does not have a done hole"
```

…because we don't have meaningful exception types in this module anyway.